### PR TITLE
fix(rich-text-input): focus hijack prevention

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

We had some bugs to do with focus hijacking with the `RichTextInput`. This PR fixes though, and adds a new Formik example for the Rich Text Input

### CODE REVIEW

File: src/components/inputs/rich-text-input/editor.js [hunk 1]
```diff
@@ -29,14 +32,23 @@ const Editor = props => {
...
+  }, [props.editor.value.document, updateRenderToggleButton]);
 
   return (
     <CollapsibleMotion
       minHeight={COLLAPSED_HEIGHT}
       isDefaultClosed={!props.defaultExpandMultilineText}
     >
       {({ isOpen, toggle, containerStyles, registerContentNode }) => {
+        // opens the input if it regains focus and it's closed
+        if (prevIsFocused !== props.isFocused && props.isFocused && !isOpen) {
```
Comment (montezume) [inline src/components/inputs/rich-text-input/editor.js]: so if the focus has changed from not being focused to focused, and it's not open, we should toggle it open

File: src/components/inputs/rich-text-input/rich-text-input.js [hunk 2]
```diff
@@ -21,17 +25,36 @@ class RichTextInput extends React.PureComponent {
     this.props.onChange(event);
   };
 
+  onBlur = (event, editor, next) => {
+    next();
```
Comment (montezume) [inline src/components/inputs/rich-text-input/rich-text-input.js]: these are the fixes for the onBlur / onFocus.

File: src/components/internals/rich-text-body/rich-text-body.js [hunk 3]
```diff
@@ -98,82 +99,88 @@ const RichTextEditorBody = React.forwardRef((props, ref) => {
   const hasUndos = props.editor.hasUndos();
   const hasRedos = props.editor.hasRedos();
 
-  const hasBlock = type =>
-    props.editor.value.blocks.some(node => node.type === type);
+  const onClickBlock = React.useCallback(
```
Comment (montezume) [inline src/components/internals/rich-text-body/rich-text-body.js]: just added a useCallback for this big function

File: src/components/internals/rich-text-body/rich-text-body.js [hunk 4]
```diff
@@ -98,82 +99,88 @@ const RichTextEditorBody = React.forwardRef((props, ref) => {
...
+  );
 
-  const onChangeMoreStyles = val => {
-    if (!props.editor.value.selection.isFocused) {
-      props.editor.focus();
-    }
-    props.editor.toggleMark(val.value);
-  };
+  const onChangeMoreStyles = React.useCallback(
+    val => {
```
Comment (montezume) [inline src/components/internals/rich-text-body/rich-text-body.js]: same thing here

File: src/components/internals/rich-text-body/rich-text-body.js [hunk 5]
```diff
@@ -215,7 +222,7 @@ const RichTextEditorBody = React.forwardRef((props, ref) => {
             styles={tooltipStyles}
           >
             <Button
-              isActive={props.editor.hasUnderlinedMark()}
+              isActive={props.isFocused && props.editor.hasUnderlinedMark()}
```
Comment (montezume) [inline src/components/internals/rich-text-body/rich-text-body.js]: now that we are tracking the focus, we fix a UX problem and only display active buttons when the editor is in focus.

**Comment by montezume** [comment]
Abel is sick and Tobi is on vacation, so I tagged @amine-benselim 😇

File: src/components/inputs/rich-text-input/rich-text-input.js (hunk 2 above)
Comment (amine-benselim) [reply src/components/inputs/rich-text-input/rich-text-input.js]: maybe referencing the issue for why we are using `setTimeout` is going to be helpful 🤔

**Pull request merged**